### DISCUSSION TITLE
replacing modemu with modemu2k

### DISCUSSION THREAD

**Discussion by andy5995** [dosemu2/dosemu2#1482] [General]
I've done a little work on modemu and created a fork called [modemu2k](https://github.com/theimpossibleastronaut/modemu2k) a few years ago. One of its [chief differences](https://github.com/theimpossibleastronaut/modemu2k/blob/master/ChangeLog) is that it's ipv6-capable now.

Whether or not it's worth the effort to replace the plugin you're using now is another matter. ;)

**Comment by stsp** [comment]
We need your work, if:
- You can address this request:
https://www.spinics.net/lists/linux-msdos/msg02126.html
i.e. have the inbound connections feature.
- You can walk through all the mods
we did (like non-blocking connect etc)
and get them in.
- Probably make it a library for simpler
integration.

What do you think?

**Reply by andy5995** [reply]
> We need your work, if:
> 
>     * You can address this request:
>       https://www.spinics.net/lists/linux-msdos/msg02126.html
>       i.e. have the inbound connections feature.

I created a [ticket for myself](https://github.com/theimpossibleastronaut/modemu2k/issues/32)

> 
>     * You can walk through all the mods
>       we did (like non-blocking connect etc)
>       and get them in.

What mods? Modifications to modemu? I already altered the network code in modemu2k to make it ipv6 capable, but I'll take a look at your local files.

> 
>     * Probably make it a library for simpler
>       integration.

Most of the source code files compile into a static library now. Not in the last release, but the git version.

**Comment by stsp** [comment]
Also see this:
https://www.spinics.net/lists/linux-msdos/msg02128.html
It discusses projects like tcpser and tty0tty.

**Comment by andy5995** [comment]
You can [see here](https://github.com/theimpossibleastronaut/modemu2k/blob/102d6bba0c01694c304199d1da3bacf7a7f670ba/src/sock.c#L64-L97) I've replaced gethostbyname() with getaddrinfo(). From the gethostbyname() man page

>The gethostbyname*(), gethostbyaddr*(), herror(), and hstrerror() functions  are
 obsolete.   Applications should use getaddrinfo(3), getnameinfo(3), and gai_str‐
error(3) instead.

But I haven't played around yet with any non-blocking code. I'll see how far I get.

**Comment by stsp** [comment]
> You can see here I've replaced gethostbyname() with getaddrinfo().

If you want to submit this as a
patch to our code - no problems
at all.
What I meant before, is that "we
need your code as a replacement
of our existing code, if this and that".
But if you can instead provide the
patches to our code, then there is
no need to implement the incoming
connections or whatever. Just submit
whatever mods you have, and we'll
review.

**Comment by andy5995** [comment]
@stsp After 5 years... :) I can say that modemu2k now supports inbound connections (https://github.com/theimpossibleastronaut/modemu2k/issues/67), and it is a library, and non-blocking connections have been implemented. However, I haven't verified that all your mods are in place. But this is where it's at. Hopefully in the future we'll have some time to do some testing. Or we'll just leave things as they are if your modemu patches are doing what you and your users need them to do. :) But do let me know if you're interested and I'll see if I (or Claude) can verify how your modemu is modded and see if modemu2k would be a good fit. Cheers!

**Comment by stsp** [comment]
Good to hear that!
Its not necessary to make sure
all my patches are applied.
What matters is if you can use
some terminal app to atd to an
IP address and exchange data.
So will you prepare the patch
for dosemu?

**Reply by andy5995** [reply]
https://github.com/dosemu2/dosemu2/pull/2925
